src/Docs/CLI/Evaluate.hs

```diff
@@ -1046,20 +1046,33 @@ packageModuleUrl (PackageUrl purl) moduleName =
     stripSuffix x s = maybe s reverse $ stripPrefix x $ reverse s
 
 -- =============================
---  HTTP requests
+--  Fetch HTML
 -- =============================
 
+-- | Fetch and cache request's content
 fetchHTML :: HasUrl a => a -> M HtmlPage
 fetchHTML x = do
-  req <- Http.parseRequest (getUrl x)
-  src <- fetch req
+  cache <- State.gets sCache
+  let uri = getUrl x
+  src <- cached cache uri $ fetch uri
   return (parseHtmlDocument src)
 
--- | Fetch and cache request's content
-fetch :: Http.Request -> M LB.ByteString
-fetch req = do
-  cache <- State.gets sCache
-  cached cache (show req) $ do
+-- | Decide how to fetch depending on the URI
+fetch :: Url -> M LB.ByteString
+fetch uri | "https://" `isPrefixOf` uri 
```

Comment:
We could separate the concerns of this check a little bit by using a new datatype and a conversion function.

```haskell
data Location
  = Local FilePath
  | Remote Url

location :: Url -> Maybe Location
```

Reply:
I am not completely sure how you would do this, so maybe you could just do this, if you think it would be good.